lang/cs/Org.Apache.REEF.IO/PartitionedData/FileSystem/FilePartition.cs

```diff
@@ -0,0 +1,94 @@
+﻿/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Org.Apache.REEF.IO.FileSystem;
+using Org.Apache.REEF.IO.FileSystem.Hadoop;
+using Org.Apache.REEF.IO.PartitionedData.FileSystem.Parameters;
+using Org.Apache.REEF.IO.PartitionedData.Random.Parameters;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Utilities.Logging;
+
+namespace Org.Apache.REEF.IO.PartitionedData.FileSystem
+{
+    internal sealed class FilePartition<T> : IPartition<IEnumerable<T>>
```

Comment:
@dkm2110 you mean Dispose? Destructor is more or less an anti-pattern in C#.
@jwang98052 for file based system, what does row even mean? Aren't Streams more common pattern there? You would arguably not read one row at a time from remote file anyways. Would an abstraction like this make more sense?
```c#
public interface IResultSet<T>
{
    bool HasMore {get;}
    IEnumerable<T> Rows {get;}
    IResultSet GetNextResultSet();
}
```